### ISSUE TITLE
mod_offline does not work properly with MS SQL server

### ISSUE THREAD

**Issue by mrfrostfire** [processone/ejabberd#1007]
Hi,

I'm setting up an ejabberd instance where we are required to have MS SQL as backing end.
I have installed 16.02 on our RHEL 7 servers using the rpm package and set up the conf file properly for using the mssql server as database.
Everything works as it should except for one thing, whenever I set mod_offline to use odbc as backing store or set default_db to odbc, offline messages stop working properly.

The setup we have works like this:
1. A user is generated.
2. A muc is set up
3. An invite to the muc is sent to the user
4. The user connects to the ejabberd server, gets the invite and joins the muc.

Unfortionally, there is no errors whatsoever in the logs, and the row is created in the mssql database properly in spool as seen below (verified in ssms):
2016-03-14 12:00:36.365 [debug] <0.316.0>@ejabberd_odbc:sql_query_internal:474 SQL: "select count(*) from spool where username='xxxxxxxx';"
2016-03-14 12:00:36.366 [debug] <0.319.0>@ejabberd_odbc:sql_query_internal:474 SQL: "begin;"
2016-03-14 12:00:36.367 [debug] <0.319.0>@ejabberd_odbc:sql_query_internal:474 SQL: "insert into spool(username, xml) values ('xxxxxxxx', '<message from=''e14cffa17b5b44009de788d268ab36f1@conference.yyy.com'' to=''smmtest01@devagent.plumchoice.com''><x xmlns=''jabber:x:conference'' jid=''e14cffa17b5b44009de788d268ab36f1@conference.zzz.com'' reason=''1''/><delay xmlns=''urn:xmpp:delay'' from=''zzz.com'' stamp=''2016-03-14T16:00:36.365Z''>Offline Storage</delay></message>');"
2016-03-14 12:00:36.370 [debug] <0.319.0>@ejabberd_odbc:sql_query_internal:474 SQL: "commit;"

But when the user connects, that offline message is just wiped away.
2016-03-14 12:01:00.213 [debug] <0.411.0>@ejabberd_c2s:presence_update:2038 from unavail = true
2016-03-14 11:25:30.125 [debug] <0.316.0>@ejabberd_odbc:sql_query_internal:474 SQL: "begin;"
2016-03-14 11:25:30.126 [debug] <0.316.0>@ejabberd_odbc:sql_query_internal:474 SQL: "select username, xml from spool where username='xxxxxxxx'  order by seq;"
2016-03-14 11:25:30.127 [debug] <0.316.0>@ejabberd_odbc:sql_query_internal:474 SQL: "delete from spool where username='xxxxxxxx';"
2016-03-14 11:25:30.129 [debug] <0.316.0>@ejabberd_odbc:sql_query_internal:474 SQL: "commit;"

After that, there's nothing but the regular presence, roster etc exchange.

**Comment by badlop** [comment]
It works correctly for me using ejabberd 16.02, MySQL 5.5.44

If you configure other modules to use ODBC, like mod_roster, mod_last or mod_vcard, is the information delivered back correctly, or it is also lost?

How did you configure the storage? I set this
```
odbc_type: mysql
odbc_server: "localhost"
odbc_database: "ejabberd_test"
odbc_username: "ejabberd_test"
odbc_password: "ejabberd_test"
```

**Comment by mrfrostfire** [comment]
That is MySQL, my issue is with MS SQL.

Everything else seems to work, and the row is actually populated in the
table, but then gets deleted without being processed properly.

tors 17 mars 2016 kl 12:46 skrev badlop <notifications@github.com>:

**Comment by zinid** [comment]
This should be fixed by 7a9e93839a9ff8f2cb098fcbd75d2669520521bb and b0ef3e66a89a5bcf2383458d5d2af4af98568a95
